clippy_lints/src/default.rs

```diff
@@ -290,3 +291,16 @@ fn field_reassigned_by_stmt<'tcx>(this: &Stmt<'tcx>, binding_name: Symbol) -> Op
         }
     }
 }
+
+/// Returns whether `expr` is the update syntax base: `Foo { a: 1, .. base }`
+fn is_update_syntax_base<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) -> bool {
+    if_chain! {
+        if let Some(parent) = get_parent_expr(cx, expr);
+        if let ExprKind::Struct(_, _, Some(base)) = parent.kind;
+        then {
+            base.hir_id == expr.hir_id
```

Comment:
This looks weird to me. Is this the correct way if two expressions are the same?

Reply:
Depends on what you want to do. Since you want to check that those two expressions are _exactly_ the same (in the same position and the same "content"), yes comparing `HirId` is the way to go.

FYI: If you want to compare 2 expressions if they "look" the same, we have `SpanlessEq` in `clippy_utils` that does that for you.